### PR TITLE
BC-1951 Add info to text topic

### PR DESCRIPTION
**Pull request by bischofmax**
# Description

## Links to Tickets or other pull requests

[CKEditor Doku for Plugins](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/tutorials/implementing-a-block-widget.html#creating-a-command)

https://github.com/hpi-schul-cloud/dof_app_deploy/pull/324
https://github.com/hpi-schul-cloud/schulcloud-server/pull/3591
https://github.com/hpi-schul-cloud/schulcloud-client/pull/2944

## Approval for review

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

File: static/scripts/topicEdit.js [hunk 1]
```diff
@@ -493,7 +501,18 @@ class TopicText extends TopicBlock {
      */
 	render() {
 		const editorId = (this.props.content || {}).editorId || this.editorId;
-		return (
+		const infoBox = <div class="alert info-custom">
+							<div className="fa fa-info-circle" />
+							Dateien können erst nach dem ersten Speichern des Themas hochgeladen werden.
```
Comment (bischofmax) [inline static/scripts/topicEdit.js:506]: Text internationalisieren

File: controllers/topics.js [hunk 2]
```diff
@@ -274,16 +276,12 @@ router.post('/', async (req, res, next) => {
...
-	}).then(() => {
+	}).then((lesson) => {
 		if (req.body.referrer) {
 			res.redirect(`${(req.headers.origin)}/${req.body.referrer}`);
 		}
-		res.redirect(
-			context === 'courses'
-				? `/courses/${req.params.courseId
-				}${req.query.courseGroup ? `/groups/${req.query.courseGroup}` : '/?activeTab=topics'}`
-				: `/teams/${req.params.teamId}/?activeTab=topics`,
```
Comment (bischofmax) [inline controllers/topics.js:285]: @SevenWaysDP Aus meiner Sicht gibt es keine Möglichkeit mehr in Teams eine Lesson zu erstellen. Deswegen entfällt die Möglichkeit hier. Courses und Coursegroups sind beide durch die Lösung abgedeckt.

File: controllers/topics.js [hunk 3]
```diff
@@ -274,16 +276,12 @@ router.post('/', async (req, res, next) => {
...
-		);
+
+		res.redirect(`${(req.headers.origin)}/courses/${req.params.courseId}/topics/${lesson._id}/edit`);
```
Comment (CeEv) [inline controllers/topics.js:284]: How it handle lessons in coursegroups?

https://default-main.cd.dbildungscloud.dev/courses/62eb686155c2e66cb2c5d4ba/topics/62f226c8c239f2001d5d28f4?courseGroup=62f226b7c239f2001d5d2747

Teacher login

File: static/styles/topics/topics.scss [hunk 4]
```diff
@@ -14,3 +16,17 @@
...
+	color: $fontColor;
+	display: flex;
+	flex-direction: row;
+	align-items: center;
+	gap: 18px;
+
+	i {
+		margin: auto;
+	}
+}
```
Comment (CeEv) [inline static/styles/topics/topics.scss:32]: github linter: 
line break on the end of the file

**Review by CeEv — CHANGES_REQUESTED** [review]

File: controllers/topics.js (hunk 3 above)
Comment (bischofmax) [reply controllers/topics.js]: The '?courseGroup='-Param doesn't makes a difference on the edit page. On the detail page it adds the hierarchy navigation on top. But for consistency reasons I added it with 2a15438009c584db1a7c61dd6dc227eb99879335
FYI @csp175

File: static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/file-browser-helper.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+const apiV3BasePath = '/api/v3';
+
+export default class FileBrowserHelper {
+	static async getFileUrl() {
+		const topicId = $('#content-blocks').data('topicid');
+		const schoolId = $('#content-blocks').data('schoolid');
+		const courseFileUrl = $('#url-input').val();
+
+		if (topicId !== undefined && schoolId !== undefined) {
+			return this.copyAsLessonFile(schoolId, 'lessons', topicId, courseFileUrl);
```
Comment (SevenWaysDP) [inline static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/file-browser-helper.js:12]: Please change 'lessons' read from data.

**Review by SevenWaysDP — CHANGES_REQUESTED** [review]

File: static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/file-browser-helper.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
+import getCookie from '../../../../helpers/cookieManager';
+
+const apiV3BasePath = '/api/v3';
```
[the remark this answers is not in the document]
Comment (bischofmax) [reply static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/file-browser-helper.js:3]: const url = `${apiBasePath}/download/${fileRecordId}/${fileName}`;

[the remark this answers is not in the document]
Comment (bischofmax) [reply static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/file-browser-helper.js:3]: Helper funktion in storage.js

File: static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/file-browser-helper.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+		const courseFileUrl = $('#url-input').val();
+
+		if (topicId !== undefined && schoolId !== undefined) {
+			return this.copyAsLessonFile(schoolId, 'lessons', topicId, courseFileUrl);
+		}
+
+		return courseFileUrl;
+	}
+
+	static async copyAsLessonFile(schoolId, parentType, parentId, url) {
```
Comment (bischofmax) [inline static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/file-browser-helper.js:18]: Rename to copyFile

**Review by SevenWaysDP — APPROVED** [review]

**Review by CeEv — APPROVED** [review]

**Pull request merged**